### PR TITLE
PE-169 : [xmp-config.xml] - migrate xmp-config.xml to the new CM

### PR DESCRIPTION
**Pull request by psuthar-opennms**
### All Contributors

* [ ] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [ ] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://issues.opennms.org) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the JIRA issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here,
or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/PE-169

### CODE REVIEW

File: features/config/upgrade/src/main/resources/changelog-cm/29.0.0/changelog-cm.xml [hunk 1]
```diff
@@ -24,10 +24,12 @@
         <cm:importConfig schemaId="discovery" configId="default" filePath="discovery-configuration.xml"/>
     </changeSet>
 
-    <changeSet author="dherdt" id="1.0-register-jmxconfig">
```
Comment (freddy61025) [inline features/config/upgrade/src/main/resources/changelog-cm/29.0.0/changelog-cm.xml:27]: you didn't do the rebase properly.

Comment (freddy61025) [reply features/config/upgrade/src/main/resources/changelog-cm/29.0.0/changelog-cm.xml:27]: Some of changes are removed by you branch.

File: opennms-config/src/main/resources/META-INF/opennms/applicationContext-commonConfigs.xml [hunk 2]
```diff
@@ -89,6 +89,7 @@
 
   <onmsgi:service interface="org.opennms.netmgt.config.api.DiscoveryConfigurationFactory" ref="discoveryFactory" />
 
+
   <bean id="databaseSchemaConfigFactory-init" class="org.springframework.beans.factory.config.MethodInvokingFactoryBean">
```
Comment (freddy61025) [inline opennms-config/src/main/resources/META-INF/opennms/applicationContext-commonConfigs.xml:93]: why add an extra line here?

File: protocols/xmp/src/main/java/org/opennms/netmgt/protocols/xmp/collector/XmpCollector.java [hunk 3]
```diff
@@ -95,6 +95,7 @@
 import org.opennms.netmgt.rrd.RrdRepository;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
```
Comment (freddy61025) [inline protocols/xmp/src/main/java/org/opennms/netmgt/protocols/xmp/collector/XmpCollector.java:98]: remove unused import

File: protocols/xmp/src/main/java/org/opennms/netmgt/protocols/xmp/config/XmpPeerFactory.java [hunk 4]
```diff
@@ -80,8 +83,8 @@ public static synchronized void init() throws FileNotFoundException, IOException
         }
         
         // Someday we might do something substantial here
-        XmpConfigFactory.init();
-        m_config = XmpConfigFactory.getInstance().getXmpConfig();
+        //XmpConfigFactory.init();
```
Comment (freddy61025) [inline protocols/xmp/src/main/java/org/opennms/netmgt/protocols/xmp/config/XmpPeerFactory.java:86]: remove comments

File: protocols/xmp/src/main/java/org/opennms/netmgt/protocols/xmp/detector/XmpDetector.java [hunk 5]
```diff
@@ -76,6 +77,9 @@
  */
 
 public class XmpDetector implements SyncServiceDetector {
+    @Autowired
+    XmpConfigFactory xmpConfigFactory;
+
```
Comment (freddy61025) [inline protocols/xmp/src/main/java/org/opennms/netmgt/protocols/xmp/detector/XmpDetector.java:82]: should define private

File: protocols/xmp/src/main/java/org/opennms/netmgt/protocols/xmp/detector/XmpDetector.java [hunk 6]
```diff
@@ -145,8 +149,8 @@ public void onInit() {
         try {
             XmpConfig protoConfig;
 
-            XmpConfigFactory.init();
-            protoConfig = XmpConfigFactory.getInstance().getXmpConfig();
+            //XmpConfigFactory.init();
```
Comment (freddy61025) [inline protocols/xmp/src/main/java/org/opennms/netmgt/protocols/xmp/detector/XmpDetector.java:152]: comments

File: protocols/xmp/src/main/java/org/opennms/netmgt/protocols/xmp/monitor/XmpMonitor.java [hunk 7]
```diff
@@ -59,7 +60,8 @@
  * @version $Id: $
  */
 public class XmpMonitor extends AbstractServiceMonitor {
-
+    @Autowired
+    XmpConfigFactory xmpConfigFactory;
     /**
```
Comment (freddy61025) [inline protocols/xmp/src/main/java/org/opennms/netmgt/protocols/xmp/monitor/XmpMonitor.java:65]: private

File: protocols/xmp/src/test/java/org/opennms/netmgt/protocols/xmp/collector/XmpCollectorComplianceTest.java [hunk 8]
```diff
@@ -44,6 +44,7 @@
 import org.opennms.netmgt.snmp.InetAddrUtils;
 
 import com.google.common.collect.ImmutableMap;
+import org.springframework.beans.factory.annotation.Autowired;
```
Comment (freddy61025) [inline protocols/xmp/src/test/java/org/opennms/netmgt/protocols/xmp/collector/XmpCollectorComplianceTest.java:48]: unused

**Review by freddy61025 — CHANGES_REQUESTED** [review]
some code cleanup and recover deleted entry.

**Review by psuthar-opennms — COMMENTED** [review]
All review comments implemented

**Review by freddy61025 — APPROVED** [review]

**Pull request merged**